neuromation/cli/formatters/ftable.py

```diff
@@ -0,0 +1,104 @@
+from enum import Enum
+from itertools import zip_longest
+from textwrap import wrap
+from typing import Any, Iterator, List, Optional, Sequence, Union
+
+
+__all__ = ["table"]
+__version__ = "0.1"
+
+
+class Align(str, Enum):
+    CENTER = "center"
+    LEFT = "left"
+    RIGHT = "right"
+
+
+def table(
+    rows: Sequence[Sequence[str]],
+    widths: Sequence[Optional[Union[int, range]]] = (),
+    aligns: Sequence[Align] = (),
+    max_width: Optional[int] = None,
+) -> Iterator[str]:
+
+    # Columns widths calculation
+    if len(widths) < len(rows[0]):
+        widths = tuple(widths) + tuple([None]) * (len(rows[0]) - len(widths))
+    calc_widths: List[int] = []
+    for i in range(len(widths)):
+        width = widths[i]
+        if isinstance(width, int):
+            calc_widths.append(width)
+            continue
+        max_cell_width: int = max(len(row[i]) for row in rows)
+        if width is None:
+            calc_widths.append(max_cell_width)
+        elif isinstance(width, range):
+            if max_cell_width in width:
+                calc_widths.append(max_cell_width)
+            else:
+                calc_widths.append(width[-1])
+        else:
+            raise TypeError(f"Unsopported width[{i}]: {widths[i]}")
+
+    # How many empty columns can be displayed
+    max_empty_columns = len(rows[0])
+    if max_width:
+        for i in range(len(widths)):
+            if sum(calc_widths[0 : (i + 1)]) > max_width:
+                max_empty_columns = i
+                break
```

Comment:
It is not a large problem, because we are not going to format many thousands of columns, but this loop has a quadratic complexity. A loop that has a linear complexity:
```python
        sum_width = 0
        for i, width in enumerate(calc_widths):
            sum_width += width
            if sum_width > max_width:
                max_empty_columns = i
                break
```